application/main.go

```diff
@@ -405,6 +407,12 @@ func main() {
 		logger.Fatalf("failed to parse app config: %v", err)
 	}
 
+	if conf.CacheExpirationTime != nil {
+		regManager.LivenessTester = new(lt.CachedLivenessTester)
+		regManager.LivenessTester.Init(conf.CacheExpirationTime)
+		regManager.IsCached = true
+	}
+
```

Comment:
Couple of things because this is the place where a lot of the code goes into action. `conf.CacheExpirationTime` is type string so it isn't nil-able so we need to compare to an empty string here (default when unset). 

Second, we can remove the `IsCached` var since we are relying on the interface (change suggested in the review comment for `lib/registration.go`). 

Next we should throw an error here if there is an issue parsing the duration in `Init()`  (change suggested in the review comment for `liveness/liveness.go`). 

Finally we can initialize the CLT before assigning it so that the compiler doesn't try to call `Init()`  on the LivenessTester interface which the compiler wont allow. 

```go 
	if conf.CacheExpirationTime != "" {
		clt := &lt.CachedLivenessTester{}
		err = clt.Init(conf.CacheExpirationTime)
		if err != nil {
			logger.Fatal(err)
		}
		regManager.LivenessTester = clt
	}
```